workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -24,8 +24,81 @@ namespace <#= qualifiedNamespace #>
         {
             public uint ComponentId => <#= componentDetails.ComponentId #>;
 
-            public BlittableBool DirtyBit { get; set; }
-<# foreach(var fieldDetails in fieldDetailsList) { #>
+            private BlittableBool isDirty;
+
+            // Bit masks for tracking which component properties were changed locally and need to be synced.
+            // Each byte tracks 8 component properties.
+<# for (var i = 0; i < (fieldDetailsList.Count / 8) + 1; i++) { #>
+            private byte dirtyBits<#= i #>;
+<# } #>
+
+            public bool IsDirty()
+            {
+                return isDirty;
+            }
+
+            /*
+            The propertyIndex arguments starts counting from 0. It depends on the order of which you defined
```

Comment:
"The propertyIndex arguments starts counting from 0. It depends on the order of which you defined your component properties in a schema component but is not the schema field number itself."
->
"The propertyIndex arguments counts up from 0, in the order defined in the schema component. It is *not* the schema field number itself."

"e.g." -> "For example,"

Reply:
Fixed!